### PR TITLE
Adds Notification List on HomePage with sorting and filtering options.

### LINKED ISSUE #30: Home Screen is empty.
**Current Behaviour**
Home screen shows nothing for now.

**Expected Behaviour**
I think there should be a list of all the notifications on the home screen with an option to filter w.r.t group and also an option to sort according to our need.

### PR DESCRIPTION
**Pull request by DreamyPhobic**
Closes #30

### CODE REVIEW

**Comment by jbeorse** [comment]
Does this show all messages from all groups, or does it pre-filter based on the groups that the ODK user belongs to? For example, imagine there are groups: north_clinic, east_clinic. Imagine we have users north_admin (part of north_clinic group), east_admin(part of east_clinic group), and north_east_director (part of north_clinic and east_clinic groups).

If I'm logged in as north_clinic I would expect to only see messages for the north_clinic group (and all other groups messages would be inaccessible to me). If I'm logged in as east_clinic I would expect to only see messages for east_clinic group. If I'm logged in as north_east_director I would expect to see messages from both groups. 

Does this PR add the option for noth_east_director to filter based on the multiple groups that he is a part of?

**Comment by DreamyPhobic** [comment]
Yes initially it will show all the messages of groups which are subscribed by that user. And by clicking on filter, user can filter messages by group.

**Review by jbeorse — APPROVED** [review]

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -18,4 +18,11 @@
     <string name="error">Error</string>
     <string name="json_file_missing_error_message">Server has not been configured correctly. Please, check the server settings in ODK Services.</string>
 
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (linl33) [inline app/src/main/res/values/strings.xml]: This value is not used.

**Review by linl33 — COMMENTED** [review]
The fragments for sort and filter cause a crash when Android needs to tear down and recreate them, for example on screen rotation. The problem is that they have a non zero-argument constructor, fragments cannot have these constructors. See [Fragment documentation](https://developer.android.com/reference/android/app/Fragment) for more information. To supply parameters to the fragments, they should be supplied with the `setArgument` method. See the `DetailsFragment` [here](https://developer.android.com/guide/components/fragments#Example) for an example provided by Google. It uses a static method in the fragment to ensure all parameters are set.

The strings added should be moved to `strings.xml`

2 more things that are nice to have but they shouldn't block this merge.
 - The search function launches a new instance of MainActivity, so each search creates an entry on the backstack. This is counter intuitive and generally not how this kind of search is implemented. @jbeorse what do you think the desired behavior should be?
 - It would be nice if the sort menu can remember the selected option.

**Comment by DreamyPhobic** [comment]
@linl33 @jbeorse I have implemented the search functionality according to the developer guide https://developer.android.com/guide/topics/search/search-dialog. 
And I think it's not creating any extra instance of MainActivity, I have also checked with the android profiler.

**Review by linl33 — APPROVED** [review]

**Pull request merged**